R/g.sib.sum.R

```diff
@@ -17,16 +17,31 @@ g.sib.sum = function(SLE,M,ignorenonwear=TRUE,desiredtz="") {
   colnames(sleep) = colnames(A)[(which(colnames(A)=="night")+1):ncol(A)]
   ws3 = M$windowsizes[1]
   ws2 = M$windowsizes[2]
-  sib.cla.sum = as.data.frame(matrix(0, 1000,9))
+  sib.cla.sum = as.data.frame(matrix(-1, 1000,9))
+  # label night after last night as -1 to differentiate it from night 0
+  lastNightEnd = max(which(night == max(night)))
+  if (length(night) > lastNightEnd) {
+    night[(lastNightEnd + 1):length(night)] = -1
+  }
+  # unique nights
   un = unique(night) #unique nights
-  missingnights = which(un == 0 | is.na(un) == TRUE)
+  missingnights = which(un == -1 | is.na(un) == TRUE)
   if (length(missingnights) > 0) {
     un = un[-missingnights]
   }
+  # if recording starts after 4am, then also remove first night
+  first4am = grep("04:00:00", M$metalong$timestamp)[1]
+  firstmidnight = grep("00:00:00", M$metalong$timestamp)[1]
+  if (firstmidnight < first4am) { # then recording started after 4am and first night is not calculated
```

Comment:
Can line 33-35 not be simplify by simplify checking whether the first timestamp is later in the day than 4am?
This would avoid having to apply grap on all timestamps.

Reply:
I have tried to implement this but I think there should be an easier way to do it, this is what I did:

```
  firstTS = M$metashort$timestamp[1]
  firstTS_posix = iso8601chartime2POSIX(firstTS, tz = desiredtz)
  firstday4am = paste0(substr(firstTS, 1, 11), "04:00:00", substr(firstTS, 20, nchar(firstTS)))
  firstday4am_posix = iso8601chartime2POSIX(firstday4am, tz = desiredtz)
  if (firstday4am_posix < firstTS_posix) { # then first timestamp is later than 4am
```

Reply:
How about:
```
firstTS_posix = iso8601chartime2POSIX(M$metashort$timestamp[1], tz = desiredtz)
firstday4am_posix = as.POSIXct(paste0(as.Date(firstTS_posix), " 04:00:00"), tz = desiredtz)
if (firstday4am_posix < firstTS_posix) { # then first timestamp is later than 4am
```

Reply:
Done, and I've tested that it works.